tests/test_codesigned.py

```diff
@@ -0,0 +1,98 @@
+STUB_FOLDER = REPO_ROOT / "conda" / "shell"
+
+
+@lru_cache(maxsize=None)
+def find_signtool() -> str | None:
+    """Tries to find signtool
+
+    Prefers signtool on PATH otherwise searches system
+    """
+    signtool_path = which("signtool")
+    if signtool_path:
+        return signtool_path
+
+    # Common installation directories where signtool might be located
+    common_paths = [
+        "C:\\Program Files (x86)\\Windows Kits\\10\\bin",
+        "C:\\Program Files\\Windows Kits\\10\\bin",
+        "C:\\Windows\\System32",
+    ]
+
+    signtool_path = None
+    # Search for signtool in common paths
+    for path in common_paths:
+        if signtool_path:
+            # We found one already
+            return signtool_path
+        if not os.path.exists(path):
+            continue
+        signtool_path = os.path.join(path, "signtool.exe")
+        if os.path.exists(signtool_path):
+            return signtool_path
+        elif "Windows Kits" in path:
+            signtool_path = None
+            max_version = 0
+            for dirname in os.listdir(path):
+                # Use most recent signtool version
+                if not dirname.endswith(".0"):
+                    continue  # next dirname
+                if int(dirname.replace(".", "")) < max_version:
+                    continue  # next dirname
+
+                maybe_signtool_path = os.path.join(path, dirname, "x64", "signtool.exe")
+                if os.path.exists(maybe_signtool_path):
+                    signtool_path = maybe_signtool_path
+    return signtool_path
+
+
+@lru_cache(maxsize=None)
+def signtool_unsupported_because() -> str:
+    reason = ""
+    if not on_win:
+        reason = "Only verifying signatures of stub exe's on windows"
+        return reason
+    signtool = find_signtool()
+    if not signtool:
+        reason = "signtool: unable to locate signtool.exe"
+    try:
+        check_output([signtool, "verify", "/?"])
+    except CalledProcessError as exc:
+        reason = f"signtool: something went wrong while running 'signtool verify /?', output:\n{exc.output}\n"
+    return reason
+
+
+def signtool_unsupported() -> bool:
+    return True if signtool_unsupported_because() else False
+
+
+@pytest.mark.skipif(signtool_unsupported(), reason=signtool_unsupported_because())
+@pytest.mark.parametrize("stub_file_name", ["cli-32.exe", "cli-64.exe"])
+def test_stub_exe_signatures(stub_file_name: str) -> None:
+    """Verify that signtool verifies the signature of the stub exes"""
+    logger = logging.getLogger(__name__)
+    stub_file = STUB_FOLDER / stub_file_name
+    logger.info(f"Checking for signature in {stub_file}")
+    signtool_exe = find_signtool()
+    try:
+        # raises CalledProcessError if this fails
+        check_call([signtool_exe, "verify", "/pa", "/v", stub_file])
+    except CalledProcessError as e:
+        logger.error(f"Command failed with exit code {e.returncode}")
+        raise
```

Comment:
```suggestion
    completed_process = subprocess.run([signtool_exe, "verify", "/pa", "/v", stub_file])
    assert completed_process.return_code == 0
```

I think this would be a little more concise. We're also making use of assertions here rather than just raising an error. This should be a little easier to debug in pytest results in the future in case this test begins failing.